step/step.py

```diff
@@ -162,35 +164,58 @@ def run(self, **kwargs):
 
     def pull(
         self,
-        save_dir: Union[str, Path] = "",
         data_version: Optional[str] = None,
         bucket: Optional[str] = None,
     ):
         # Resolve None bucket
         if bucket is None:
-            bucket = self._storage_bucket
+            bucket = self.storage_bucket
 
         # Resolve save_dir
         save_dir = file_utils.resolve_directory(save_dir, make=True)
 
-        # Run pull for each upstream
+        # Resolve the project and branch for use in quilt paths
+        repo = git.Repo(Path(".").expanduser().resolve())
+        current_branch = repo.active_branch.name
+        package_name = self.__module__.split(".")[0]
+
+        # Run checkout for each upstream
         for upstream_task in self.upstream_tasks:
-            pass
+            upstream_task.checkout(data_version=data_version, bucket=bucket)
```

Comment:
We can't really test this now but I have a feeling this will break because each push to a sub module will have a different version and the user may need to provide a dictionary mapping step name to version if they don't just want the latest